services/galley/test/integration/API/Teams/LegalHold.hs

```diff
@@ -785,18 +829,181 @@ testNoConsentBlockDeviceHandshake = do
+                rel @?= Conn.MissingLegalholdConsent
+              _ -> assertBool "wrong event type" False
+
+        forM_ [(legalholder, peer), (peer, legalholder)] $ \(one, two) -> do
+          putConnection one two Conn.Accepted
+            !!! testResponse 403 (Just "bad-conn-update")
+
+        assertConnections legalholder [ConnectionStatus legalholder peer Conn.MissingLegalholdConsent]
+        assertConnections peer [ConnectionStatus peer legalholder Conn.MissingLegalholdConsent]
+
+        -- TODO: test if message sending is blocked
+        -- for_ (mbConn >>= Conn.ucConvId) $ \convId -> do
+        -- (again, other label / 4xx status code would also be fine.)
+        -- postOtrMessageJson peer convId !!! testResponse 412 (Just "missing-legalhold-consent")
+        -- postOtrMessageProto peer convId !!! testResponse 412 (Just "missing-legalhold-consent")
+
+        do
+          doDisableLH
+          assertConnections
+            legalholder
+            [ ConnectionStatus legalholder peer $
+                if testPendingConnection then Conn.Sent else Conn.Accepted
+            ]
+          assertConnections
+            peer
+            [ ConnectionStatus peer legalholder $
+                if testPendingConnection then Conn.Pending else Conn.Accepted
+            ]
+
+        -- TODO: test if message sending works again
+        -- postOtrMessageJson undefined undefined !!! const 201 === statusCode
+        pure ()
+      else do
+        doEnableLH
+        postConnection legalholder peer !!! do testResponse 412 (Just "missing-legalhold-consent")
+        postConnection peer legalholder !!! do testResponse 412 (Just "missing-legalhold-consent")
 
 testNoConsentBlockGroupConv :: TestM ()
 testNoConsentBlockGroupConv = do
   -- "If LH is activated for other user in group conv, this user gets removed with helpful message"
   -- tracked here: https://wearezeta.atlassian.net/browse/SQSERVICES-428
   pure ()
 
+testBenchHack :: HasCallStack => TestM ()
+testBenchHack = do
+  {- representative sample run on an old laptop:
+
+     (10,0.186728036s)
+     (30,0.283852693s)
+     (100,0.712145446s)
+     (300,1.72513614s)
+     (600,3.47943481s)
+
+     the test itself is running for ages, but most of the time is spent in setting up the
+     connections.
+
+     before running this test, you also need to change {galley,brig}.integration.yaml:
+
+     ```
+       diff --git a/services/brig/brig.integration.yaml b/services/brig/brig.integration.yaml
+       -  setUserMaxConnections: 16
+       -  setMaxTeamSize: 32
+       -  setMaxConvSize: 16
+       +  setUserMaxConnections: 999
+       +  setMaxTeamSize: 999
+       +  setMaxConvSize: 999
+       diff --git a/services/galley/galley.integration.yaml b/services/galley/galley.integration.yaml
+       -  maxTeamSize: 32
+       -  maxFanoutSize: 18
+       -  maxConvSize: 16
+       +  maxTeamSize: 999
+       +  maxFanoutSize: 999
+       +  maxConvSize: 999
```

Comment:
I think I don't understand why conversation size and team size has to be increased, if this is about connections, which usually occur between users of different teams? (And cannot occur between users of the same team). Shouldn't this bench also work just with an increase of setUserMaxConnections and maybe maxFanoutSize?

Reply:
It might, I was just increasing values until it worked.

Reply:
da45b70af1feee8317d6ff314bed693e945e54a8